### PR TITLE
Fixed #13497 returning "empty" values as `null`

### LINKED ISSUE #13497: [BUG] Libmemcached returns null, if the recovered data is empty
### Expected and Actual Behavior
Libmemcached returns null if the recovered data is empty. 

Libmemcached should returns the data even if they empty. 

> Provide minimal script to reproduce the issue

```php
<?php
$mc = new Memcached();

$mc->addServers(array(
    array('127.0.0.1',11211),
));

$lm = new \Phalcon\Cache\Backend\Libmemcached(new \Phalcon\Cache\Frontend\None(), [
    'servers' => [
        [
            'host' => '127.0.0.1',
            'port' => 11211,
            'weight' => 0,
        ],
    ]
]);

$array = [];

$lm->save('empty.array', $array, 1);

echo PHP_EOL . 'empty.array' . PHP_EOL;
echo 'libmemcached: ';
var_dump($lm->get('empty.array'));
echo 'memcached:    ';
var_dump($mc->get('empty.array'));


$lm->save('empty.int', 0, 1);

echo PHP_EOL . 'empty.int' . PHP_EOL;
echo 'libmemcached: ';
var_dump($lm->get('empty.int'));
echo 'memcached:    ';
var_dump($mc->get('empty.int'));


$lm->save('empty.string', '', 1);

echo PHP_EOL . 'empty.string' . PHP_EOL;
echo 'libmemcached: ';
var_dump($lm->get('empty.string'));
echo 'memcached:    ';
var_dump($mc->get('empty.string'));

$lm->save('empty.bool', false, 1);

echo PHP_EOL . 'empty.bool' . PHP_EOL;
echo 'libmemcached: ';
var_dump($lm->get('empty.bool'));
echo 'memcached:    ';
var_dump($mc->get('empty.bool'));
```

> output 
```
empty.array
libmemcached: NULL
memcached:    array(0) {
}

empty.int
libmemcached: NULL
memcached:    int(0)

empty.string
libmemcached: NULL
memcached:    string(0) ""

empty.bool
libmemcached: NULL
memcached:    bool(false)
```

### Details

* Phalcon version: 3.4.1
* PHP Version: 7.2.9
* Operating System: Debian 9
* Installation type: package manager
* Other related info : 
- memcached : 1.4.33
- php_memcached : 3.0.4

### PR DESCRIPTION
**Pull request by CameronHall**
Hello!

* Type: bug fix
* Link to issue: #13497

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change: `libmemcached` no longer incorrectly returns "empty" values as null

Thanks

### CODE REVIEW

File: phalcon/mvc/model/query.zep [hunk 1]
```diff
@@ -3325,7 +3325,7 @@ class Query implements QueryInterface, InjectionAwareInterface
 			}
 
 			let result = cache->get(key, lifetime);
-			if result !== null {
+			if !empty result {
```
Comment (niden) [inline phalcon/mvc/model/query.zep]: Sorry to be a pain but this needs to be in its own PR since it doesn't relate to this one. This way we will know what changed when.

Comment (niden) [reply phalcon/mvc/model/query.zep]: Can we also write a test or two for this PR?

**Review by niden — CHANGES_REQUESTED** [review]

File: phalcon/mvc/model/query.zep (hunk 1 above)
Comment (CameronHall) [reply phalcon/mvc/model/query.zep]: I can do that, but I felt it was related as the cache can return "falsey" values. Now that the cache adapters can return `""`, `false` and `null` it seemed like an appropriate "catch-all" because it won't simply return `null` anymore.

I'll get some tests sorted too.

Comment (niden) [reply phalcon/mvc/model/query.zep]: Thanks bud

**Comment by CameronHall** [comment]
This is ready for review @niden

**Comment by niden** [comment]
@CameronHall Can I be a pain one more time please?

- Rebase your work. There is a new section in the changelog where you can add your change.
- Can you move the test to the relevant test stub in the `Cache` subfolder? Since this is `I get empty value as null` then my choice would be `Cache\Backend\Libmemcached\GetCest`. Create a new method there using the same naming as the one in there (small variation) and add the `@issue` in the docblock along with your name and email/github if you wish. 

The `LibmemcachedCest.php` file will be refactored at some point so I want to not add to them. Also you can use the `PhalconLibmemcached` module to check memcached without using Phalcon and also you can create a trait under `_data/fixtures/Traits/Cache` to allow you to connect to libmemcached (same as the `getCache()` method).

**Comment by niden** [comment]
@CameronHall I will fix the phpcs issues

File: CHANGELOG-4.0.md [hunk 2]
```diff
@@ -126,6 +126,7 @@
 - Changed the `Phalcon\Session` namespace by refactoring the component. `Phalcon\Session\Manager` is now the single component offering session manipulation by using adapters. Each adapter implements PHP's `SessionHandlerInterface`. Available adapters are `Phalcon\Session\Files`, `Phalcon\Session\Libmemcached`, `Phalcon\Session\Noop` and `Phalcon\Session\Redis`.  [#12833](https://github.com/phalcon/cphalcon/issues/12833), [#11341](https://github.com/phalcon/cphalcon/issues/11341), [#13535](https://github.com/phalcon/cphalcon/issues/13535)
 - Fixed `Phalcon\Mvc\Models` magic method (setter) is fixed for arrays  [#13661](https://github.com/phalcon/cphalcon/issues/13661)
 - Fixed `Phalcon\Mvc\Model::skipAttributes` and `Phalcon\Mvc\Model::allowEmptyColumns` allowEmptyStrings & skipAttributes repsect the column mapping. [#12975](https://github.com/phalcon/cphalcon/issues/12975), [#13477](https://github.com/phalcon/cphalcon/issues/13477) 
+- Fixed `\Phalcon\Cache\Backend\Libmemcached` returning "empty" values being as `null` when they could be `0`, `false` or empty `string`. [#13497](https://github.com/phalcon/cphalcon/issues/13497)
```
Comment (sergeyklay) [inline CHANGELOG-4.0.md]: @CameronHall, @niden why do you use previous version fro this change? Did you see `[4.0.0-alpha.2]` at the top of this  file?

Comment (CameronHall) [reply CHANGELOG-4.0.md]: That is my mistake @sergeyklay. I rebased the changelog without checking the output/changes. I'll fix it in another PR.

Comment (CameronHall) [reply CHANGELOG-4.0.md]: Looks like it's been fixed anyway. Thanks @niden.
